[bugfix] Hide speaker dropdown for safari browser

edwardlee-msft opened the pull request:
# What
<!--- Describe your changes. -->
In configuration page, hide speaker dropdown for safari browser
![image](https://github.com/Azure/communication-ui-library/assets/97130533/bffdd2f5-50d3-4a69-967c-53c4581b3ec9)


# Why
<!--- What problem does this change solve? -->
<!--- Provide a link if you are addressing an open issue. -->
https://skype.visualstudio.com/SPOOL/_workitems/edit/3398753
![image](https://github.com/Azure/communication-ui-library/assets/97130533/17c5f962-82a3-4c17-9a6c-41f94376ed2a)


# How Tested
<!--- How did you test your change. What tests have you added. -->
Locally on MacOs Safari browser. Call Composite.

# Process & policy checklist
<!--- Review the list and check the boxes that apply. -->

- [ ] I have updated the project documentation to reflect my changes if necessary.
- [ ] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.
<!--- If yes, describe the impact. -->

JamesBurnside commented:
Hmm I suppose we should remove this empty space for this scenario
![image](https://github.com/Azure/communication-ui-library/assets/2684369/cee6d0c7-31de-4732-a90c-1bf3243e1225)

edwardlee-msft commented:
> Hmm I suppose we should remove this empty space for this scenario -- might need @pereiralex input ![image](https://private-user-images.githubusercontent.com/2684369/294056673-cee6d0c7-31de-4732-a90c-1bf3243e1225.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQzOTI2MDEsIm5iZiI6MTcwNDM5MjMwMSwicGF0aCI6Ii8yNjg0MzY5LzI5NDA1NjY3My1jZWU2ZDBjNy0zMWRlLTQ3MzItYTkwYy0xYmYzMjQzZTEyMjUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDEwNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAxMDRUMTgxODIxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZTNjZjNhM2MwZjMyZDFhZDYyYmUzMzYzODcyOTRmMGY3OGRjNjIyODA5OWE4YTkxNTUxZmU4MjIyMjcwZGRmNCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.xB9W7eh49wYChc-u6I7PvbBLzJeGv7i5RJqE3KVit50)

Confirmed with Alex P that we should shrink the container size. 
New commit addresses those issues. Please refer to the updated screenshot provided in the description of this pr.

JamesBurnside commented on packages/react-composites/src/composites/CallComposite/styles/CallConfiguration.styles.ts:
```diff
@@ -91,10 +91,10 @@ export const configurationSectionStyle: IStackStyles = {
  * @private
  */
-export const selectionContainerStyle = (theme: ITheme): string =>
+export const selectionContainerStyle = (theme: ITheme, browser?: string): string =>
```
nit: difficult to understand why safari means the height is different, suggestion: use something like `noSpeakerDropdownShown` as the variable name here

JamesBurnside commented on packages/react-composites/src/composites/CallComposite/components/LocalDeviceSettings.tsx:
```diff
@@ -117,6 +117,8 @@ export const LocalDeviceSettings = (props: LocalDeviceSettingsType): JSX.Element
 
   const cameraPermissionGranted = props.cameraPermissionGranted;
   const micPermissionGranted = props.microphonePermissionGranted;
+  const isSafariBrowser = adapter.getState().environmentInfo?.environment.browser.toLowerCase() === 'safari';
```
Could this instead be isSafariBrowser **&& speakers.length === 0**? My worry is one day safari will suddenly expose changing speakers independently

JamesBurnside reviewed (approved):
Couple suggestions - looks good though!
